utils/vlog/tests/carry/golden.pb_type.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version='1.0' encoding='utf-8'?>
+<pb_type xmlns:xi="http://www.w3.org/2001/XInclude" name="CARRY" num_pb="1">
+  <input name="CIN" num_pins="1"/>
```

Comment:
Missing `<pack_pattern type="carry"/>`;
```xml
  <input name="CIN" num_pins="1"/>
    <pack_pattern name="C" type="carry"/>
  </input>
```

This should be implicitly discovered because the `carry` attribute from the `CBLOCK` object should be propagated outwards.

Reply:
I don't think we should implicitly discover carry here (and in general in those cases). I have implemented the code doing that (see the patch https://pastebin.com/zCKvuGPX it will work only with #997). This indeed adds carry pack_pattern to CIN and COUT. However, VPR fails to parse the test architecture with the following error:

```
Error 1:                                                                       
Type: Architecture file                                                        
File: /tmp/really-really-really-really-really-really-really-really-really-really-really-really-really-long-path/conda/conda-bld/vtr_1567590526578/work/vpr/src/pack/prepack.cpp
Line: 179                                                                      
Message: Failed to find root block for pack pattern carry-C
```

This error makes sense because adding pack pattern definition to CIN and COUT ports here makes the carry chain span from the device input to output, so effectively the root block of it is an input buffer. VPR cannot handle this case. If we leave the automatic discovery if will always propagate the carry chains up to the top level block. I think we should stay with explicit carry_chains definitions.

Reply:
Looking at the diagram below;

<img src="https://docs.google.com/drawings/d/e/2PACX-1vQZSLI_sGwBvfDY4QugJafRvIySP-PA6IFLPHTKa99Bfxo2aeBud4Ioo_ypP02cAKR8PMkW-nVF5xlV/pub?w=2034&amp;h=1659">

It is my understanding that all the red arrows in `Figure - A2` should have the `pack_pattern` marked?